ddtrace/profiling/_threading.pyx

```diff
@@ -1,52 +1,64 @@
 from __future__ import absolute_import
 
-import threading
+import sys
+import threading as ddtrace_threading
 import typing
 import weakref
 
 import attr
-
-from ddtrace.internal import nogevent
+from six.moves import _thread
 
 
 cpdef get_thread_name(thread_id):
-    # This is a special case for gevent:
-    # When monkey patching, gevent replaces all active threads by their greenlet equivalent.
-    # This means there's no chance to find the MainThread in the list of _active threads.
-    # Therefore we special case the MainThread that way.
-    # If native threads are started using gevent.threading, they will be inserted in threading._active
-    # so we will find them normally.
-    if thread_id == nogevent.main_thread_id:
-        return "MainThread"
-
-    # We don't want to bother to lock anything here, especially with eventlet involved 😓. We make a best effort to
-    # get the thread name; if we fail, it'll just be an anonymous thread because it's either starting or dying.
-    try:
-        return threading._active[thread_id].name
-    except KeyError:
+    # Do not force-load the threading module if it's not already loaded
+    if "threading" not in sys.modules:
+        return None
+
+    import threading
+
+    # Look for all threads, including the ones we create
+    for threading_mod in (threading, ddtrace_threading):
+        # We don't want to bother to lock anything here, especially with
+        # eventlet involved 😓. We make a best effort to get the thread name; if
+        # we fail, it'll just be an anonymous thread because it's either
+        # starting or dying.
         try:
-            return threading._limbo[thread_id].name
+            return threading_mod._active[thread_id].name
         except KeyError:
-            return None
+            try:
+                return threading_mod._limbo[thread_id].name
+            except KeyError:
+                pass
+
+    return None
```

Comment:
Looking at [ddtrace/profiling/_threading.pyi](https://github.com/DataDog/dd-trace-py/blob/4dc15e2722b8aec02aa81d50bbdd40e91232ba4e/ddtrace/profiling/_threading.pyi#L4) get_thread_name() should return a string. Here we return None.  Is this something we should address (this existed before, I just noticed now)?

Reply:
Good catch. We should fix this, but perhaps in a separate PR.